### ISSUE TITLE
php-fpm8.1 crashed with SIGSEGV in spx_reporter_full_get_key()

### ISSUE THREAD

**Issue by stdex** [NoiseByNorthwest/php-spx#183]
```
DistroRelease: Ubuntu 20.04
ExecutablePath: /usr/sbin/php-fpm8.1

ThreadStacktrace:
Thread 1 (Thread 0x7f585de6d5c0 (LWP 53294)):
#0  0x00007f585b401ae4 in spx_reporter_full_get_key () from /usr/lib/php/20210902/spx.so
#1  0x00007f585b3fe822 in profiling_handler_start.part () from /usr/lib/php/20210902/spx.so
#2  0x00007f585b3feaaf in profiling_handler_init () from /usr/lib/php/20210902/spx.so
#3  0x00007f585b3fda24 in zm_activate_spx () from /usr/lib/php/20210902/spx.so
#4  0x000055d2bb226f60 in zend_activate_modules ()
#5  0x000055d2bb1bc9ec in php_request_startup ()
#6  0x000055d2bb06589a in ?? ()
#7  0x00007f585fef80b3 in __libc_start_main (main=0x55d2bb0649e0, argc=4, argv=0x7ffe4395b1e8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffe4395b1d8) at ../csu/libc-start.c:308
#8  0x000055d2bb06670e in _start ()
```

On apache+mod_php work correct, on nginx+php_fpm crash

spx.so enabled,
jit disabled,
process.dumpable = yes

**Comment by NoiseByNorthwest** [comment]
The SIGSEGV may be fixed by #184.
However the profiling may not work since the internal profiler object's creation seems to fail in your case.
One reason I see is that SPX cannot open (for writing) its data file.

**Comment by stdex** [comment]
After restart it started working.
Perhaps there is really something with write file.
